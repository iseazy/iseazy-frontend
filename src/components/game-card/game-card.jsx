export const GameCard = ({ id, src, flipped, title, onClick }) => {
    const style = {
        backgroundImage: `url(${ flipped ? src : 'images/background.svg' })`
    };
    const className = `game-play__card ${ flipped ? 'game-play__card--flipped' : 'game-play__card--enabled' }`;
    const handleClick = () => onClick(id);

    return (
        <div className="game-play-card__wrap">
            <button className={ className } style={ style } onClick={ handleClick }>
                <h1>{ flipped ? '' : title }</h1>
            </button>
        </div>
    );
};
