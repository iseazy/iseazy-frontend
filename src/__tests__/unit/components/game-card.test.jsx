import { render, fireEvent, screen } from '@testing-library/react';
import { GameCard } from '../../../components/game-card/game-card';

describe('<GameCard />', () => {
    const testProps = {
        id: 1,
        src: 'testSource',
        flipped: true,
        title: '0',
        onClick: () => {}
    };
    
    it('should render flipped card', () => {
        const { asFragment } = render(<GameCard { ...testProps }/>);
        expect(asFragment()).toMatchSnapshot();
    });
    
    it('should render unflipped card', () => {
        const { asFragment } = render(<GameCard { ...testProps } flipped={ false }/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should be clicked', () => {
        const onClick = jest.fn();
        render(<GameCard { ...testProps } onClick={ onClick }/>);

        fireEvent.click(screen.getByRole('button'));

        expect(onClick).toHaveBeenCalledWith(testProps.id);
    });
});
