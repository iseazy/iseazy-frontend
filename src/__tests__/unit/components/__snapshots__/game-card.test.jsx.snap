// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameCard /> should render flipped card 1`] = `
<DocumentFragment>
  <div
    class="game-play-card__wrap"
  >
    <button
      class="game-play__card game-play__card--flipped"
      style="background-image: url(testSource);"
    >
      <h1 />
    </button>
  </div>
</DocumentFragment>
`;

exports[`<GameCard /> should render unflipped card 1`] = `
<DocumentFragment>
  <div
    class="game-play-card__wrap"
  >
    <button
      class="game-play__card game-play__card--enabled"
      style="background-image: url(images/background.svg);"
    >
      <h1>
        0
      </h1>
    </button>
  </div>
</DocumentFragment>
`;
